src/components/views/right_panel/UserInfo.tsx

```diff
@@ -1434,18 +1462,29 @@ const BasicUserInfo: React.FC<{
     let verifyButton;
     const homeserverSupportsCrossSigning = useHomeserverSupportsCrossSigning(cli);
 
-    const userTrust = cryptoEnabled && cli.checkUserTrust(member.userId);
-    const userVerified = cryptoEnabled && userTrust && userTrust.isCrossSigningVerified();
+    const userTrust = useAsyncMemo<UserVerificationStatus | undefined>(
+        async () => cli.getCrypto()?.getUserVerificationStatus(member.userId),
+        [member.userId],
+        // the user verification status is not initialized
+        undefined,
+    );
+    const hasUserVerificationStatus = Boolean(userTrust);
+    const isUserVerified = Boolean(userTrust?.isVerified());
     const isMe = member.userId === cli.getUserId();
     const canVerify =
-        cryptoEnabled && homeserverSupportsCrossSigning && !userVerified && !isMe && devices && devices.length > 0;
+        hasUserVerificationStatus &&
+        homeserverSupportsCrossSigning &&
+        !isUserVerified &&
+        !isMe &&
+        devices &&
+        devices.length > 0;
 
     const setUpdating: SetUpdating = (updating) => {
         setPendingUpdateCount((count) => count + (updating ? 1 : -1));
     };
     const hasCrossSigningKeys = useHasCrossSigningKeys(cli, member as User, canVerify, setUpdating);
 
-    const showDeviceListSpinner = devices === undefined;
+    const showDeviceListSpinner = !hasUserVerificationStatus || devices === undefined;
```

Comment:
arg. This means we will show an infinite spinner if `getCrypto()` is undefined?

Reply:
Is it possible to have uninitialized crypto here ? Crypto is initialized right after login, right ?

Reply:
good question.

> Crypto is initialized right after login, right ?

Usually, yes, but in theory it is possible to have a version of element that doesn't support crypto (see condition at https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/MatrixClientPeg.ts#L310). You used to be able to do that by not shipping the `olm.js` file, though I'm not sure if that still works. The whole thing was because of export regulations (you weren't allowed to distribute "secure encryption" to countries that the western governments don't like), though I think we may have quietly given up on that.

So, in theory, it's possible to have uninitialized crypto here. Whether that's true in practice, I don't know.

Reply:
Hmm okay, I changed it to handle the uninitialized crypto